percona/percona-server-mongodb-operator
# K8SPSMDB-185: Allow PBM to get S3 credentials from EC2 metadata

[![K8SPSMDB-185](https://badgen.net/badge/JIRA/K8SPSMDB-185/green)](https://jira.percona.com/browse/K8SPSMDB-185) [<img width="16" alt="Powered by Pull Request Badge" src="https://user-images.githubusercontent.com/1393946/111216524-d2bb8e00-85d4-11eb-821b-ed4c00989c02.png">](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

To test:

Create an EKS cluster using a similar config to:

```yaml
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: ege-psmdb-cluster
  region: eu-west-1
  version: "1.21"
nodeGroups:
  - name: ng-1
    minSize: 3
    maxSize: 5
    iam:
      attachPolicyARNs:
      - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
      - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
      - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
      - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
      - arn:aws:iam::aws:policy/AmazonS3FullAccess
    instancesDistribution:
      maxPrice: 0.15
      instanceTypes: ["m5.xlarge", "m5.2xlarge"] # At least two instance types should be specified
      onDemandBaseCapacity: 0
      onDemandPercentageAboveBaseCapacity: 50
      spotInstancePools: 2
    tags:
      'iit-billing-tag': 'jenkins-eks
```

The important part is `attachPolicyARNs`, don't change its values except `arn:aws:iam::aws:policy/AmazonS3FullAccess` (we can test it with different policies).

Afterwards configure backup storage in CR but leave `credentialsSecret` empty. PBM will use the EC2 metadata to authenticate requests.
